src/App.js

```diff
@@ -24,20 +26,22 @@ class App extends Component {
   componentDidMount = async() => {
     const data = await fetchData(this.state.query, this.state.direction);
     this.setState({data})
+    this.setState({loading:false})
   }
   
   render() { 
-    const { load } = this.state;
-    const filterType = this.data.filter(
-      (pokemon) => {
-        return (this.data.type_1 === 'All' || this.data.type_1 === pokemon.type_1)
-      }
-    )
-    return ( 
-      <>
+    const {data} = this.state;
+    const {loading} = this.state;
+    const filterType = data.filter(
+      (pokemon) => 
+         pokemon.typeOptionList === this.state.typeOptionList || this.state.typeOptionList === data
+      
+      )
+      return ( 
+        <>
      <h4>Pokemon! Gotta Catch Them All!</h4>
-     {load && <h2>Please Wait</h2>}
-     {!load && (
+     {loading && <image url='' >Please Wait</image>}
```

Comment:
Also, remember that you want an actual loading spinner animation rather than  simply the words "Please Wait" as this provides a better UX for the user.